src/Foundation/Http/Middleware/CheckForTrustedHost.php

```diff
@@ -0,0 +1,82 @@
+<?php namespace October\Rain\Foundation\Http\Middleware;
+
+use Config;
+use Illuminate\Http\Middleware\TrustHosts as BaseMiddleware;
+
+class CheckForTrustedHost extends BaseMiddleware
+{
+    /**
+     * Get the host patterns that should be trusted.
+     *
+     * Trusted hosts should be defined in the `config/app.php` configuration file as an array, ie.:
+     *
+     *   'trustedHosts' => [
+     *      'example.com',           // Matches just example.com
+     *      'www.example.com',       // Matches just www.example.com
+     *      '^(.+\.)?example\.com$', // Matches example.com and all subdomains
+     *      'https://example.com',   // Matches just example.com
+     *   ],
+     *
+     * or as a boolean - if true, it will trust the `app.url` host and all subdomains, if false it
+     * will disable the feature entirely.
+     *
+     * Hosts can be defined as regex patterns for complex matching.
+     *
+     * @return array
+     */
+    public function hosts()
+    {
+        return self::processTrustedHosts(Config::get('app.trustedHosts', []));
+    }
+
+    /**
+     * Processes the trusted hosts into an array of patterns the match for host header checks.
+     *
+     * @param array|bool $hosts
+     * @return array
+     */
+    public static function processTrustedHosts($hosts)
+    {
+        if ($hosts === true) {
+            $url = Config::get('app.url', null);
+
+            // If no app URL is set, then disable trusted hosts.
+            if (is_null($url)) {
+                return [];
+            }
+
+            // If app.url is used, allow both the domain and the `www` subdomain
+            $host = parse_url($url, PHP_URL_HOST);
+
+            if (preg_match('/^www\.(.*?)$/i', $host, $matches)) {
+                return '^(www\.)?' . preg_quote($matches[1]) . '$';
+            } else {
+                return '^(www\.)?' . preg_quote($host) . '$';
```

Comment:
Shouldn't these be `$host =` instead of `return` given that the return type is an array? Also, could you add an inline comment to explain why there's two different branches here? (if I'm not mistaken, something like `// Ensure that www.example.com and example.com app.url values are normalized to allow both www.example.com and example.com`

Reply:
@LukeTowers does the comment here suffice? https://github.com/octobercms/library/pull/549/files/2cc51c9f523a24c5e2aabb54ce86eb754c69a0e6#diff-25fa1eb570e4c381198dcffea34a2c648e9d3ad5d05afe7a657d156fd120d7b0R48